docs/contributor/arch/001-otel-collector-metric-based-pipeline-status.md

```diff
@@ -0,0 +1,102 @@
+
+#### Direct Prometheus Queries:
+* The operator manages a set of predefined PromQL queries and communicates with Prometheus using [expression queries API](https://prometheus.io/docs/prometheus/latest/querying/api/#expression-queries) to collect data points.
+  Subsequently, it utilizes this data to evaluate the status of Pipeline CRs. 
+* However, it's crucial to note that if the querying process takes an extended duration, it may impact the reconciliation of TracePipeline and MetricPipeline.
+![Prometheus Integration using Direct Queries](../assets/prom-integration-direct-queries-flow.svg "Prometheus Integration using Direct Queries")
+
+#### Prometheus Queries with Additional Controller:
+* Similar to [Direct Prometheus Queries](#direct-prometheus-queries), but it addresses the potential challenge of long-running queries, which could hinder reconciliation introducing a separate PipelineMonitoring CR with its own reconciliation loop.
+  This separation effectively isolates the query execution and result storage from the TracePipeline and MetricPipeline controllers.
+  It's worth noting that while this approach enhances efficiency, it can introduce increased complexity to the overall setup.
+![Prometheus Integration with Additional Controller](../assets/prom-integration-extra-ctrl-flow.svg "Prometheus Integration with Additional Controller")
+
+#### Prometheus Alerts:
+* In this approach, PromQL expressions are defined as Prometheus alerts, with Prometheus responsible for their evaluation.
+* The operator leverages the [alerts API](https://prometheus.io/docs/prometheus/latest/querying/api/#alerts) to periodically retrieve information regarding active alerts.
+* This method offers several advantages, including the avoidance of potentially long-running queries that can hinder reconciliation, as the evaluation is offloaded to Prometheus.
+* Additionally, it provides better troubleshooting capabilities as Prometheus state can be observed using the dashboard.
+* Prometheus can also push alerts, allowing the operator to serve a webhook that, when called, triggers a reconciliation (see [channel watch](https://github.com/kubernetes-sigs/controller-runtime/blob/818a2e12a70c7a6d263edd180695e4448d2de823/pkg/source/example_test.go#L42))
+![Prometheus Integration using Alerts](../assets/prom-integration-alerts-flow.svg "Prometheus Integration using Alerts")
+
+## Decision
+
+### Metric Analysis
+Despite its inherent complexity, our preference leans toward running Prometheus as a standalone deployment. This approach offers the highest level of flexibility and ensures future-proofing, making it the most robust choice for our needs.
+The following things have to be considered:
+* To maintain security, we should make sure only the operator can access Prometheus, and we can do this by deploying a NetworkPolicy.
+* For cost-effectiveness and faster queries, it's a good idea for Prometheus to use in-memory storage.
```

Comment:
Not sure if Prometheus can run "in-memory" only. I guess using ephemeral storage (e.g., an emptyDir volume) for the TSDB is what we want here.